### ISSUE TITLE
consider renaming have.js_property to have.property

### ISSUE THREAD

**Issue by yashaka** [yashaka/selene#538]
will this even work for mobile? o_O
if .get_property is valid for mobile
then we should rename it for sure here...

**Comment by ka7e-se** [comment]
Yes, the .get_property method is implemented in Appium, and it provides a way to access the properties of elements on the web page or mobile application that are accessible via JavaScript properties. One of the main differences between .get_property and .get_attribute in Appium is the type of information they retrieve from the elements.

**1. .get_property in Appium:**
 - Allows you to access the properties of elements that are accessible through JavaScript properties.
- Useful for fetching dynamic values like CSS property values, innerText, and other properties not directly available through standard HTML attributes.
- Provides a way to interact with elements based on their properties that are not easily accessed using traditional attribute methods.
Ex: `element.get_property('checked')` to get the state of a checkbox.

**2. .get_attribute in Appium:**
- Used to retrieve the values of specific HTML attributes of elements, such as id, class, href, src, etc.
- Provides access to the attribute values of elements that are defined in the HTML structure.
- Typically used for retrieving static attributes and other specific attribute values defined in the HTML.
Ex:  `element.get_attribute('id')` to extract the element's identifier.


How .get_property and .get_attribute differ from the raw Selenium implementation:

1. Appium Implementation:
- In Appium, .get_property and .get_attribute methods behave similarly to their WebDriver counterparts but are optimized for mobile automation.
- The behavior and functionality of these methods in Appium are designed to work seamlessly in mobile automation scenarios.

**2. Selenium Implementation:**
- In raw Selenium, .get_property and .get_attribute methods function similarly to Appium but are more geared towards traditional web automation.
- The implementation of these methods in Selenium WebDriver is tailored to interact with web elements in desktop browsers.

In summary, while both Appium and Selenium provide .get_property and .get_attribute methods, the main difference lies in their optimization for mobile automation in Appium and traditional web automation in Selenium. The purpose and usage of these methods remain similar, with a focus on accessing element properties and attributes for effective automation testing.

**Comment by ka7e-se** [comment]
Here's what I've managed to pull from AI-chats. Would be nice to double check the examples in real test runs:

**Android**
Interactions occur through UIAutomator.

- _get_attribute_: used to retrieve values of element attributes such as 'content-desc', 'text', 'resource-id', and other properties defined in the Android XML markup.

- _get_property:_ used to get values that may be accessible through native platform APIs, although its usage is often less common than get_attribute.

Ex: 

```
element = driver.find_element_by_id('com.example:id/myElement')

content_desc = element.get_attribute('content-desc')
print(content_desc)

text_property = element.get_property('text')
print(text_property)

```

**iOS**
Interactions occur through XCUITest

- _get_attribute:_ used to get values of element attributes such as 'label', 'value', 'name', and other properties defined in the iOS XML markup.

- _get_property:_ As with Android, this method is less commonly used but can still be employed to retrieve native properties of elements.

Ex: 
```
element = driver.find_element_by_id('myElement')

label = element.get_attribute('label')
print(label)

value_property = element.get_property('value')
print(value_property)
```


**Difference from 'raw' Selenium Implementation**
In raw Selenium **'get_attribute'** and **'get_property'** have distinct roles:

- **get_attribute:** retrieves the value of an attribute as defined in the HTML markup.
- **get_property:** retrieves the value of a property of a DOM element, which can be different from the attribute.

Ex:
```
element = driver.find_element_by_id('myElement')

class_attr = element.get_attribute('class')
print(class_attr)

inner_text = element.get_property('innerText')
print(inner_text)

```
So:
In Appium both **'get_attribute'** and **'get_property'** interact with native properties of elements rather than through JavaScript. The distinction between them is less pronounced compared to raw Selenium, where **'get_attribute'** deals with HTML attributes and **'get_property'** deals with DOM properties. In Appium, both methods are used to retrieve information specific to the mobile platform (Android or iOS) but **'get_attribute'** is more commonly used.
